src/iterator.cr

```diff
@@ -29,6 +29,16 @@ require "./enumerable"
 # (1..10_000_000).each.select(&.even?).map { |x| x * 3 }.first(3).to_a # => [6, 12, 18]
 # ```
 #
+# Because iterators only go forward, when using methods that consume it entirely or partially -
```

Comment:
OCD: Could you use en dash please? Same in the following line.
```suggestion
# Because iterators only go forward, when using methods that consume it entirely or partially –
```

Reply:
I appreciate the OCD :). Was going to pop that in, but wasn't what the feeling towards unicode chars were.